qCC/ccComparisonDlg.cpp

```diff
@@ -940,6 +947,28 @@ bool ccComparisonDlg::computeDistances()
 				}
 			}
 			ccLog::Warning("[ComputeDistances] Result has been split along each dimension (check the 3 other scalar fields with '_X', '_Y' and '_Z' suffix!)");
+            if (mergeXY)
+            {
+                ccLog::Warning("[ComputeDistances] compute 2D distance (xy plane)");
+                int sf2D = m_compCloud->getScalarFieldIndexByName(qPrintable(m_sfName + QString(" (XY)")));
+                if (sf2D >= 0)
+                    m_compCloud->deleteScalarField(sf2D);
+                sf2D = m_compCloud->addScalarField(qPrintable(m_sfName + QString(" (XY)")));
+                CCCoreLib::ScalarField* sf = m_compCloud->getScalarField(sf2D);
+                for (int idx = 0; idx < m_compCloud->size(); idx++)
```

Comment:
theoretically you should check that 'sf' exists (just in case)

Reply:
exact